UnityEngine.Base.mm/src/patch_Resources.cs

```diff
@@ -42,7 +42,8 @@ internal sealed class patch_Resources {
         [MonoModOriginal] public static extern void OnloadAsset(Object assetToUnload);
         [MonoModOriginalName("OnloadAsset")]
         public static void UnloadAsset(Object asset) {
-            if (ETGModUnityEngineHooks.UnloadAsset(asset)) {
+            bool? overrideUnload = ETGModUnityEngineHooks.UnloadAsset?.Invoke(asset);
+            if (overrideUnload.HasValue && overrideUnload.Value) {
```

Comment:
Could be simplified as
```cs
if (ETGModUnityEngineHooks.UnloadAsset?.Invoke(asset) ?? false)
```

Where it either gives you false (don't skip) by default, or whatever the hook returns. Your method is working perfectly fine, though.